Misc Commands

xunafay opened the pull request:
pull request for #3

xunafay committed:
add stuvo command

DrunkenToast committed:
Merge branch 'master' into misc

PandaGamersHDNL committed:
Pal command

PandaGamersHDNL committed:
PAL command

PandaGamersHDNL committed:
Rooster / schedule command

DrunkenToast committed:
Changed how messages interact with prefix, added rich presence support (+config settings for it). Changed ping cmd.

DrunkenToast committed:
Merge remote-tracking branch 'origin/misc' into misc

DrunkenToast committed:
Rooster prefix fix

xunafay commented on cfg/config.json:
```diff
@@ -1,7 +1,44 @@
+
+            "richpresence":
+            {
+                "statusType": "2",
+                "interval": "60000",
```
can be of data type int

xunafay commented on src/discord.ts:
```diff
@@ -7,6 +7,26 @@ export async function buildClient(): Promise<Client> {
+      Rich presence updating.
+      Value may not be below 15000 (rate-limit discord api).
+    */
+    const interval = (parseInt(data.discord.richpresence.interval) < 15000 ? 15000 : parseInt(data.discord.richpresence.interval));
```
use Math.max

DrunkenToast committed:
used Math.max() as xunafay suggested

DrunkenToast committed:
Changed to int

DrunkenToast committed:
Fixed formatting (was wrong on mobile)

Commit:
Merge branch 'master' into misc

xunafay committed:
feat: use database and tokenizer

xunafay committed:
fix: api url ending with / resulting in 404s

xunafay committed:
transfer custom commands to DB

xunafay committed:
feat: global commands

DrunkenToast committed:
Help command

PandaGamersHDNL committed:
Default channel basic command definition

TODO: add Default channel to DB

PandaGamersHDNL committed:
Merge branch 'misc' of https://github.com/practice-enterprise/canvas-discord-bot into misc

DrunkenToast committed:
hard coded ping with ms

DrunkenToast committed:
Removed latency as it's highly inaccurate

PandaGamersHDNL committed:
figured out how to send a message outside of the on message event
TOOK WAAAAAAAAY TOOO LONG,
maybe change the as TextChannel @line 17: ./checkreminders.ts

xunafay commented on src/checkreminders.ts:
```diff
@@ -0,0 +1,25 @@
+//const getguild = (guildID: string, client: Client) => Promise.resolve(client.guilds?.fetch(guildID));
+
+
+export function checkreminderinit(client: Client): void {
```
use camel case when naming functions

xunafay commented on src/checkreminders.ts:
```diff
@@ -0,0 +1,25 @@
+
+//import { promises } from "fs";
```
remove

xunafay commented on src/checkreminders.ts:
```diff
@@ -0,0 +1,25 @@
+
+import { Client, TextChannel } from 'discord.js';
+
+//const getguild = (guildID: string, client: Client) => Promise.resolve(client.guilds?.fetch(guildID));
```
remove

xunafay commented on src/checkreminders.ts:
```diff
@@ -0,0 +1,25 @@
+    const guildID = '780572565240414208'; //TODO get guild id from db to post in correct server
+    const channelID = '782283497582624778'; //TODO get channel of the reminder from the database
+
+    const guild = await Promise.resolve(client.guilds?.fetch(guildID));
```
`.fetch()` already returns a promise no need to wrap it in another promise

xunafay commented on src/checkreminders.ts:
```diff
@@ -0,0 +1,25 @@
+    const channelID = '782283497582624778'; //TODO get channel of the reminder from the database
+
+    const guild = await Promise.resolve(client.guilds?.fetch(guildID));
+    const channel: TextChannel = await Promise.resolve(guild.client.channels.fetch(channelID)) as TextChannel;
```
`.fetch()` already returns a promise no need to wrap it in another promise
[validate channel is a text channel](https://discord.js.org/#/docs/main/stable/class/Channel?scrollTo=type) before casting

DrunkenToast committed:
Adds support for not including multiple parameters in .body()

DrunkenToast committed:
Added notes command, embed for help

DrunkenToast committed:
format doc

DrunkenToast commented on src/commands.ts:
```diff
@@ -65,5 +67,46 @@ export const commands: Command[] = [
+          .command(guildConfig.prefix+'notes').text(': get notes from your current channel', true)
+          .command(guildConfig.prefix+'notes #channel').text(': get notes from your favourite channel', true)
+          .command(guildConfig.prefix+'notes #channel an amazing note').text(': Enter a note in a channel', true)
+          .build();
```
I don't like the else if chaining.

xunafay commented on src/commands.ts:
```diff
@@ -0,0 +1,112 @@
+        'description': commands.concat(guildConfig.commands).map( c => '`' + guildConfig.prefix + c.name + '`: ' + c.description).join('\n'),
+        'color': '43B581',
+        'url': 'http://mydoc.epic',
+        'footer': {text: 'Some commands support putting \'help\' behind it.'}};
```
do not quote keys when unnecessary
format properly

xunafay commented on src/commands.ts:
```diff
@@ -0,0 +1,112 @@
+          'footer': {text: 'For help: '+guildConfig.prefix+'notes help'}
+        };
+        return embed;
+      }
```
do not define functions within functions
format properly

xunafay reviewed (commented):
use string interpolation to improve readability

DrunkenToast committed:
getNotes() defined outside notes command.

DrunkenToast committed:
string interpolation

DrunkenToast committed:
coinflip, epic help aliases

PandaGamersHDNL committed:
changed code because reviews
-> removed unnecessary code
-> check if it is a textchannel before casting to it

PandaGamersHDNL committed:
Merge branch 'misc' of https://github.com/practice-enterprise/canvas-discord-bot into misc

xunafay committed:
feat: check aliases when validating against commands

PandaGamersHDNL committed:
added temp definition for the reminder command

PandaGamersHDNL committed:
Merge branch 'misc' of https://github.com/practice-enterprise/canvas-discord-bot into misc

PandaGamersHDNL committed:
add luxon for dates/times

PandaGamersHDNL committed:
extended the reminder check to support some format
TODO add check, Get from DB, remove if sent

PandaGamersHDNL committed:
moved reminder format check to reminder command
this makes sure we store proper dates in the DB

xunafay committed:
style: add models

xunafay committed:
feat: tokenizer recognize datetime

DrunkenToast committed:
Prefix change cmd (needs admin perm support)

DrunkenToast committed:
idem, removed unnecessary if else

PandaGamersHDNL committed:
add DB support for channel reminders

TODO user DM reminders

PandaGamersHDNL committed:
Merge branch 'misc' of https://github.com/practice-enterprise/canvas-discord-bot into misc

PandaGamersHDNL commented:
+ added some comments to tokenizer

PandaGamersHDNL committed:
-Fix dublicate axios imort

PandaGamersHDNL committed:
dateTime format doesn't work dissable may revisit or
delete. check code broken

PandaGamersHDNL committed:
fix reminder when channel introduced

PandaGamersHDNL committed:
+easier readability when commands collapsed

xunafay committed:
major refactor

extract api into services

xunafay committed:
fix: reminders that fail not deleted and wrong reminder target channel
